src/java/org/apache/cassandra/index/sai/disk/format/IndexComponent.java

```diff
@@ -31,6 +31,12 @@
      */
     META("Meta"),
 
+    /**
+     * Balanced tree written by {@code BlockBalancedTreeWriter} indexes mappings of term to one or more segment row IDs
+     * (segment row ID = SSTable row ID - segment row ID offset).
+     */
+    BALANCED_TREE("BalancedTree"),
```

Comment:
Do we need to update `V1OnDiskFormat#perColumnIndexComponents()` to include this? i.e. Return this and postings if we've got a numeric column index? Feels like this should also be failing a test, but isn't currently because `StorageAttachedIndexDDLTest#verifyRebuildCorruptedFiles()` (for example) is only lookin at literal column indexes...

Reply:
Agreed, I missed this completely. I'll make sure we have a test in place for this.